scilpy/tractograms/tractogram_operations.py

```diff
@@ -843,3 +854,246 @@ def split_sft_randomly_per_cluster(orig_sft, chunk_sizes, seed, thresholds):
+
+        for future in as_completed(futures):
+            chunk = futures[future]
+            try:
+                results = future.result()
+            except Exception as exc:
+                print(f'Generated an exception: {exc}')
+            else:
+                results = np.array(results)
+                diff_data[tuple(chunk.T)] = results[:, 0]
+                acc_data[tuple(chunk.T)] = results[:, 1]
+
+            # Update tqdm progress bar
+            progress_bar.update(len(chunk))
+
+    return diff_data, acc_data
+
+
+def tractogram_pairwise_comparison(sft_one, sft_two, mask, nbr_cpu=1,
+                                   skip_streamlines_distance=True):
+    """
+    Compute the difference between two sets of streamlines for each voxel in
+    the mask. This function uses multiprocessing to compute the difference
+    between two sets of streamlines for each voxel.
+
+    Parameters
+    ----------
+    sft_one: StatefulTractogram
+        First tractogram to compare.
+    sft_two: StatefulTractogram
+        Second tractogram to compare.
+    mask: np.ndarray
+        Mask of the data to compare (optional).
+    nbr_cpu: int
+        Number of CPU to use (default: 1).
+    skip_streamlines_distance: bool
+        If true, skip the computation of the distance between streamlines.
+        (default: True)
+
+    Returns
+    -------
+    List of np.ndarray
+        acc_norm: Angular correlation coefficient.
+        corr_norm: Correlation coefficient of density maps.
+        diff_norm: Voxelwise distance between sets of streamlines.
+        heatmap: Merged heatmap of the three metrics using harmonic mean.
+    """
+    global sft_1, sft_2
+    sft_1, sft_2 = sft_one, sft_two
+
+    sft_1.to_vox()
+    sft_2.to_vox()
+    sft_1.streamlines._data = sft_1.streamlines._data.astype(np.float16)
+    sft_2.streamlines._data = sft_2.streamlines._data.astype(np.float16)
+    dimensions = tuple(sft_1.dimensions)
+
+    global matched_points_1, matched_points_2
+    matched_points_1 = generate_matched_points(sft_1)
+    matched_points_2 = generate_matched_points(sft_2)
+
+    logging.info('Computing KDTree...')
+    global tree_1, tree_2
+    tree_1 = cKDTree(sft_1.streamlines._data)
+    tree_2 = cKDTree(sft_2.streamlines._data)
+
+    # Limits computation to mask AND streamlines (using density)
+    if not mask:
+        mask = np.ones(dimensions)
+
+    logging.info('Computing density maps...')
+    density_1 = compute_tract_counts_map(sft_1.streamlines,
+                                         dimensions).astype(float)
+    density_2 = compute_tract_counts_map(sft_2.streamlines,
+                                         dimensions).astype(float)
+    mask = density_1 * density_2 * mask
+    mask[mask > 0] = 1
+
+    logging.info('Computing correlation map...')
+    corr_data = correlation([density_1, density_2], None) * mask
+
+    logging.info('Computing TODI #1...')
```

Comment:
logging.info('Computing TODI on tractogram 1...')

(idem below for 2)

(You can kepp the # sign but I don't how to stop github to cite the 1st and 2nd PR when I do!!)